tests/schemata/date/DateFromIsoString.test.ts

```diff
@@ -0,0 +1,172 @@
+          requireTime === null ? undefined : { requireTime },
+        ),
+      )
+      describe('Decoder', () => {
+        test.each(valid)('validates valid date strings, %s', str => {
+          const result = ISODateString.Decoder.decode(str)
+          expect(result._tag).toBe('Right')
+        })
+        test.each(invalid)('invalidates invalid date strings, %s', str => {
+          const result = ISODateString.Decoder.decode(str)
+          if (result._tag === 'Right') console.log({ str, result: result.right })
+          expect(result._tag).toBe('Left')
+        })
+      })
+
+      describe('Encoder', () => {
+        test.each(valid)('encoding a decoded value yields original value', original => {
+          const roundtrip = pipe(
+            original,
+            ISODateString.Decoder.decode,
+            E.map(ISODateString.Encoder.encode),
+            E.getOrElseW(() => 'unexpected'),
+          )
+          expect(new Date(original).toISOString()).toEqual(roundtrip)
+        })
+      })
+
+      describe('Eq', () => {
+        test.each(valid)('determines two strings are equal', str1 => {
+          const guard = ISODateString.Guard.is
+          const eq = ISODateString.Eq.equals
+          const test = new Date(str1)
+          if (!guard(test)) {
+            throw new Error('Unexpected result')
+          }
+          expect(eq(test, test)).toBe(true)
+        })
+      })
+
+      describe('Guard', () => {
+        test.each(valid)('validates valid date strings, %s', str => {
+          const test = new Date(str)
+          const result = ISODateString.Guard.is(test)
+          expect(result).toBe(true)
+        })
+        test.each(invalid)('invalidates invalid date strings, %s', str => {
+          const test = new Date(str)
+          const result = ISODateString.Guard.is(test)
+          expect(result).toBe(false)
+        })
+      })
+
+      describe('TaskDecoder', () => {
+        test.each(valid)('validates valid date strings, %s', async str => {
+          const result = await ISODateString.TaskDecoder.decode(str)()
+          expect(result._tag).toBe('Right')
+        })
+        test.each(invalid)('invalidates invalid date strings, %s', async str => {
+          const result = await ISODateString.TaskDecoder.decode(str)()
+          expect(result._tag).toBe('Left')
+        })
+      })
+
+      describe('Type', () => {
+        test.each(valid)('validates valid date strings, %s', str => {
+          const result = ISODateString.Type.decode(str)
+          expect(result._tag).toBe('Right')
+        })
+        test.each(invalid)('invalidates invalid date strings, %s', str => {
+          const result = ISODateString.Type.decode(str)
+          expect(result._tag).toBe('Left')
+        })
+      })
+
+      describe('Arbitrary', () => {
+        it('generates valid ISODateString', () => {
+          validateArbitrary({ Arbitrary: ISODateString.Arbitrary }, D.Guard.date.is)
```

Comment:
ditto

Reply:
This one is using a different guard instance than arbitrary, I _think_ this one is as you described